docs/recipes/PWA.md

```diff
@@ -0,0 +1,339 @@
+    // along with other config
+    pwa: {
+      serviceWorker: true,
+      manifest: {
+        name: 'Pottery Store',
+        icons: [
+          {
+            src: 'https://drums.example.com/images/amphora-96.png',
+            sizes: '96x96',
+            type: 'image/png',
+          },
+        ],
+      },
+    },
+  };
+}
+```
+
+To read more on the `appConfig.pwa` and how you can turn off the service worker,
+[read the App Config documentation.][one-app-config]
+
+Once you've enabled the service worker and set up your manifest, you will
+load in a PWA ready One App. The service worker is installed automatically
+and the web manifest we've configured will be added to the rendered HTML.
+Under the hood, [`@americanexpress/one-service-worker`][one-service-worker]
+drives the service worker that One App is using.
+What you will get out of the box is caching for the `one-app` client, your modules,
+including language packs bundled in your package; One App also supports
+offline navigation, which means that when a user requests one of our pages when offline,
+`one-app` client and modules will both load up and it is because what we have already
+cached is given back to the browser when the network is down. Try testing what your modules
+do when offline.
+
+**Your development browser matters**
+
+After you've setup, explore around for browsers that provide tools for
+the service worker, the web manifest and the cache.
+Skip to the next paragraph when you've selected a browser.
+
+**Start**
+
+When you're ready to start, here are a few mini recipes to outfit your
+Holocron modules with PWA enhancements.
+
+## Offline Support
+
+It's important to note that with offline support, users that do encounter
+themselves offline would find parts of the page broken because our modules
+have loaded in but rely on data.
+It is the module owners responsibility to know when the network is offline;
+to respond with an informative UX or support an offline experience for your users.
+When offline, the modules will load as normal, however checks are needed for
+`offline` if we'd want to support the offline experience, or at least inform
+the user on the state of the app and network.
+The good thing is, it's easy to find out when the network is offline:
+
+```js
+const updateOnlineStatus = () => (navigator.onLine ? 'online' : 'offline');
+
+window.addEventListener('online', () => console.log('network status: "%s"', updateOnlineStatus()));
+window.addEventListener('offline', () => console.log('network status: "%s"', updateOnlineStatus()));
+```
+
+Let's make a React hook that uses both events.
+Once we know if/when we're on or off the network, we can make our modules
+ready for offline and tweak its behavior appropriately.
+
+```jsx
+import React from 'react';
+import { isOffline } from '@americanexpress/one-service-worker';
+
+function useOffline() {
+  const [offline, setOffline] = React.useState(false);
+  React.useLayoutEffect(() => {
```

Comment:
might also be worth justifying why they would need to use `useLayoutEffect`  over `useEffect`

Reply:
I added a comments in the offline example (b912ca5)